packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2606,48 +2620,34 @@ class RenderTransform extends RenderProxyBox {
 
   @override
   void paint(PaintingContext context, Offset offset) {
-    if (child != null) {
-      final Matrix4 transform = _effectiveTransform!;
-      if (filterQuality == null) {
-        final Offset? childOffset = MatrixUtils.getAsTranslation(transform);
-        if (childOffset == null) {
-          // if the matrix is singular the children would be compressed to a line or
-          // single point, instead short-circuit and paint nothing.
-          final double det = transform.determinant();
-          if (det == 0 || !det.isFinite) {
-            layer = null;
-            return;
-          }
-          layer = context.pushTransform(
-            needsCompositing,
-            offset,
-            transform,
-            super.paint,
-            oldLayer: layer is TransformLayer ? layer as TransformLayer? : null,
-          );
-        } else {
-          super.paint(context, offset + childOffset);
-          layer = null;
-        }
-      } else {
-        final Matrix4 effectiveTransform = Matrix4.translationValues(offset.dx, offset.dy, 0.0)
-          ..multiply(transform)..translate(-offset.dx, -offset.dy);
-        final ui.ImageFilter filter = ui.ImageFilter.matrix(
```

Comment:
@flar FYI. I think it is correct to change this code so that the offset flows directly to the ImageFilterLayer and the transform does not take this into account.

Reply:
The goldens seem different but my read of that issue with the rotation + offset in an image filter layer was that it was the transform render object that was incorrect

Reply:
I'm not so sure about this, I think it can probably be removed from this PR if it is incorrect